### PR TITLE
[OC-120] Registry webUI: preview route/view bug

### PR DESCRIPTION
**Pull request by nickbalestra**
Preview route didnt take into account of registry prefix for paths

### CODE REVIEW

**Comment by matteofigus** [comment]
The baseUrl should be set to contain the prefix too. Perhaps we should improve the baseUrl docs?

**Comment by nickbalestra** [comment]
Mmmm I see...
Yes totally not clear.
My personal feeling is that if you use two seperate configuration then the two things should be indipendent, no?

**Comment by nickbalestra** [comment]
I have an idea 😎 , let me rework this PR

**Comment by nickbalestra** [comment]
@matteofigus I've extended the `options-sanitizer`:
It now checks if a prefix exist (before sanitizing it) and if the baseUrl doesn't already contain the prefix. If both check succeed it it sanitize the baseUrl by appending the prefix to it.

This should avoid this being a braking change, and will make it possible to handle both cases:
1. prefix and baseUrl containing prefix
2. prefix and baseUrl without prefix

Let me know what you think

**Comment by nickbalestra** [comment]
add
- [ ] sanitize baseUrl with trailing slash as per https://github.com/opencomponents/get-started/issues/2

**Comment by nickbalestra** [comment]
@matteofigus do want to review this one more time before 🌴 🌞  🏄  🍺  🏖  ?

**Comment by nickbalestra** [comment]
@mattiaerre to get into merge mood, what about starting with this? 😎

**Comment by nickbalestra** [comment]
There are still a couple of issues that need to be addressed that seems the tests didn't cought:
if prefix && baseUrl contiene prefix, niente duplicate prefix...
I'll need to add more tests to cover also this scenario

**Comment by nickbalestra** [comment]
@mattiaerre I've rebased and added the scenario based tests for the the baseUrl/Prefix.

I'll rather keep this PR scoped on solving this very small 'bug' that we discovered with the heroku get started registry.
I therefore suggest we don't addd jest or takle other aspects of the sanitization like `beforePublish` yet. (Also no need to add jest to have snapshots as those are now avail for any test runner).

Can you open an issue for the `beforePublish`? Ffor jest we already have #473 
If Ok i'll revert the commit that added jest.

**Comment by nickbalestra** [comment]
@mattiaerre this should be now ready to go.
I'll update the OpenComponents website registry and the get-started repo once this is merged. Thanks!

File: .gitignore [hunk 1]
```diff
@@ -27,4 +27,4 @@ src/components/base-component-handlebars/_package
 src/components/base-component-jade/_package
 
 test/fixtures/test.tar.gz
-test/fixtures/targz-test
\ No newline at end of file
+test/fixtures/targz-test
```
Comment (mattiaerre) [inline .gitignore]: me love this!

File: test/unit/registry-domain-options-sanitiser.js [hunk 2]
```diff
@@ -69,4 +76,48 @@ describe('registry : domain : options-sanitiser', () => {
       });
     });
   });
+
+  describe('prefix and baseUrl sanitaziation', () => {
+    const prefixAndBaseUrlSchenario = [
```
Comment (mattiaerre) [inline test/unit/registry-domain-options-sanitiser.js]: love it!

Comment (mattiaerre) [inline test/unit/registry-domain-options-sanitiser.js]: typo `prefixAndBaseUrlSchenario` => `prefixAndBaseUrlScenarios`

File: test/unit/registry-domain-options-sanitiser.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ describe('registry : domain : options-sanitiser', () => {
   const sanitise = require('../../src/registry/domain/options-sanitiser');
 
   describe('when options is empty', () => {
```
Comment (mattiaerre) [inline test/unit/registry-domain-options-sanitiser.js]: this is no longer the description for this test is it?

File: test/unit/registry-domain-options-sanitiser.js [hunk 4]
```diff
@@ -69,4 +76,48 @@ describe('registry : domain : options-sanitiser', () => {
       });
     });
   });
+
+  describe('prefix and baseUrl sanitaziation', () => {
```
Comment (mattiaerre) [inline test/unit/registry-domain-options-sanitiser.js]: typo `sanitaziation` => `sanitization`

**Pull request merged**